src/panels/lovelace/special-rows/hui-weblink-row.ts

```diff
@@ -1,22 +1,44 @@
-import { html } from "@polymer/polymer/lib/utils/html-tag.js";
-import { PolymerElement } from "@polymer/polymer/polymer-element.js";
+import { html, LitElement } from "@polymer/lit-element";
+import { EntityRow, WeblinkConfig } from "../entity-rows/types";
+import { HomeAssistant } from "../../../types";
 
 import "../../../components/ha-icon.js";
 
-class HuiWeblinkRow extends PolymerElement {
-  static get template() {
+import { TemplateResult } from "lit-html";
+
+class HuiWeblinkRow extends LitElement implements EntityRow {
+  public hass?: HomeAssistant;
+  private _config?: WeblinkConfig;
+
+  static get properties() {
+    return {
+      _config: {},
+    };
+  }
+
+  public setConfig(config): void {
+    if (!config || !config.icon || !config.name || !config.url) {
```

Comment:
I wonder why we make icon mandatory, can't we default to one?

Same with name, if no name, use url?

Reply:
Also, we should add a type for this config.